### PR TITLE
Allow AuthnRequest based stepup requests

### PR DESCRIPTION
**Pull request by MKodde**
When the AuthnContextClassRefs contain valid LoA(s), then they are to be included in the step up decision process.

https://www.pivotaltracker.com/story/show/176960392

### CODE REVIEW

File: src/OpenConext/EngineBlock/Stepup/StepupDecision.php [hunk 1]
```diff
@@ -86,6 +86,10 @@ public function __construct(
 
     public function shouldUseStepup() : bool
     {
+        // If the highest level is 1, no step up callout is requred.
+        if ($this->getStepupLoa() && $this->getStepupLoa()->getLevel() === 1) {
+            return false;
+        }
         return ($this->spLoa || $this->idpLoa || count($this->authnRequestLoas) > 0 || count($this->pdpLoas) > 0);
```
Comment (thijskh) [inline src/OpenConext/EngineBlock/Stepup/StepupDecision.php:93]: If we've already done getStepupLoa(), can we then make test in the `return` statement much simpler?

File: src/OpenConext/EngineBlock/Stepup/StepupDecision.php [hunk 2]
```diff
@@ -72,11 +79,18 @@ public function __construct(
...
     }
 
     public function shouldUseStepup() : bool
     {
-        return ($this->spLoa || $this->idpLoa || count($this->pdpLoas) > 0);
+        // If the highest level is 1, no step up callout is requred.
+        if ($this->getStepupLoa() && $this->getStepupLoa()->getLevel() === 1) {
+            return false;
+        }
+        return ($this->spLoa || $this->idpLoa || count($this->authnRequestLoas) > 0 || count($this->pdpLoas) > 0);
```
Comment (Badlapje) [inline src/OpenConext/EngineBlock/Stepup/StepupDecision.php:93]: this return statement seems to be identical to the if statement in `$this->allowNoToken`.  Can both not be abstracted into a method that can be called?

Comment (Badlapje) [reply src/OpenConext/EngineBlock/Stepup/StepupDecision.php:93]: would also have the advantage of making the method name descriptive so it's easier to follow what's going on.

**Review by Badlapje — CHANGES_REQUESTED** [review]
I haven't tested it functionally as i was unsure how to do so.  That said: nice work!

One small remark, other than that it's good to go i think.

File: src/OpenConext/EngineBlock/Stepup/StepupDecision.php (hunk 1 above)
Comment (thijskh) [reply src/OpenConext/EngineBlock/Stepup/StepupDecision.php:93]: Or remove it.

File: src/OpenConext/EngineBlock/Stepup/StepupDecision.php (hunk 2 above)
Comment (MKodde) [reply src/OpenConext/EngineBlock/Stepup/StepupDecision.php:93]: Good call!

File: src/OpenConext/EngineBlock/Stepup/StepupDecision.php (hunk 1 above)
Comment (MKodde) [reply src/OpenConext/EngineBlock/Stepup/StepupDecision.php:93]: Yes, I think you are right. A `null` from the `getStepupLoa` method equals the return statements logical comparison. We can simply return `true` at that point.

**Review by Badlapje — APPROVED** [review]
Nice work! :shipit:

**Pull request merged**